crates/ruff_python_parser/src/lexer.rs

```diff
@@ -1114,6 +1293,111 @@ impl std::fmt::Display for LexicalErrorType {
     }
 }
 
+#[derive(Debug)]
+enum StringQuoteChar {
+    Single,
+    Double,
+}
```

Comment:
Nit: Implement `Copy` for these value enums. Rust will then tell you that it's more performant to pass `StringQuoteChar` by value rather by reference (1 byte vs 8 byte + dereferencing)